03-farcaster-architecture.md

```diff
@@ -49,71 +56,81 @@ The following table summarizes different aspects of each component.
+- User's data and instructions via a `client` (see [06. Datum & Instructions](./06-datum-and-instructions.md)), and
+- Self-produced loopback messages
+
+The Daemon must create a constrained runtime environment for executing the protocol, that only permits valid protocol transitions at all times. To achieve that a Petri net model of the protocol may be used to constrain the runtime environment that executes the user's respective swap role in the protocol (see [01. High Level Overview](./01-high-level-overview.md)), by only authorizing firing valid enabled protocol transitions.
 
-Client and daemon must have an initialization protocol allowing one or the other to recover from a past swap state.
+### Counter-party daemon communication
+
+The Daemon has a bidirectional communication channel with the swap counter-party's daemon. This inter-daemon communication is used to pass protocol messages between the swap counterparties. To bootstrap this communication channel we distinguish two phases: the negotiation and the swap phase. The negotiation phase allow users to discover swaps through public offers and connect their daemons (see [02. User Stories](./02-user-stories.md)).
+
+### Loopback: self-generated input messages
+
+The daemon may generate self-addressed messages. Those messages may be used to trigger transitions only based on the daemon's state, such as timers. Those transitions can e.g. represent the absence of counter-party daemon communication during a period of time, which may trigger the swap cancellation process.
+
+## The `syncer` component
+
+A syncer is specific to a blockchain and can handle a list of `tasks` related to it. Those `tasks` will be completed in different manners depending on the blockchain type and/or the blockchain state. Syncers allow the daemon to abstract a part of the logic needed to interact with a blockchain with a defined interface composed of `tasks` and `blockchain events` (see [05. Tasks & Blockchain Events](./05-tasks-and-events.md)).
+
+### Blockchain communication
 
-Client passes `datum` and `instruction` messages to the daemon. That is, client may at the user's discretion fire one of the possible protocol transitions at each moment in time. A subset of transitions may require transactions, signatures, or keys as input which are transmitted via `datum` messages. Upon client's message reception, daemon must undertake the actions associated with the fired transition, and update the swap state accordingly. Daemon updates client with the same type of messages: `datum` and `instruction`, to move forward with the swap protocol execution.
+A syncer can choose how to interact with its specified blockchain, this can be done solely through RPC calls or involve more advanced features such as e.g. consuming the 0MQ streams proposed by a full node. The simplest implementation of a syncer would made use of public high level blockchain API, however this would only be acceptable when no value is transacted.
```

Comment:
> The simplest implementation of a syncer would made use of public high level blockchain API, however this would only be acceptable when no value is transacted.

not convinced about the value of this sentence

Reply:
Agree ;) removing